README.md

```diff
@@ -123,6 +141,73 @@ will recursively (including subdirectories) load all the YAML files. For example
         headers:
           Content-Type: text/html
 ```
+
+### Templates
+Templates can be useful to assemble your payloads from parts
+
+```yaml
+- key: dog
+  kind: Template
+  template: >
+    <animal>dog</animal>
+
+- key: cat
+  kind: Template
+  template: >
+    <animal>cat</animal>
+
+- key: get-freds-pets
+  kind: Behavior
+  expect:
+    http:
+      method: GET
+      path: /fred
+  actions:
+    - reply_http:
+        status_code: 200
+        body: >
+          <human>
+            <name>fred</name>
+            <pets>
+              {{template "dog" .}}
+              {{template "cat" .}}
+            </pets>
+          </human>
+```
+
+### Abstract Behaviors
+Abstract Behaviors can be used to parameterize some data
+
+```yaml
+- key: fruit-of-the-day
+  kind: AbstractBehavior
+  values:
+    fruit: potato
+  expect:
+    condition: '{{.HTTPQueryString | contains .Values.day}}'
+    http:
+      method: GET
+      path: /fruit-of-the-day
+  actions:
+    - reply_http:
+        status_code: 200
+        body: '{"fruit": "{{.Values.fruit}}"}'
+
+- key: monday-fruit
+  kind: Behavior
+  extend: fruit-of-the-day
+  values:
+    day: monday
+    fruit: apple
+
+- key: tuesday-fruit
+  kind: Behavior
+  extend: fruit-of-the-day
+  values:
+    day: tuesday
+
+```
+
```

Comment:
💯 

Can I ask for one more example that uses AbstractBehavior and Template together?